board/amlogic/configs/m8b_m201_v1.h

```diff
@@ -288,8 +288,26 @@
     "loadbootenv=echo >>> Load Boot Script from mmc ${mmcbootdev}:1 <<<;ext4load mmc ${mmcbootdev}:1 0x11000000 /boot/uEnv.txt\0" \
     "importbootenv=echo >>> Importing environment from mmc ${mmcbootdev}:1 <<<;env import -t 0x11000000 ${filesize}\0" \
     "uenvbootcmd=ext4load mmc ${mmcbootdev}:1 ${loadaddr} /boot/${kernel_image};ext4load mmc ${mmcbootdev}:1 0x13000000 /boot/${ramdisk_image};bootm ${loadaddr}#${boardname} 0x13000000\0" \
+    "uenvbootcmd_append="\
```

Comment:
Before I go further reviewing this can you please add some documentation inline, so that this is easier to parse later. Especially the alignment concerns.
Also is there areason for the "run loadbootenv" part to be duplicated now?

Reply:
Also, does this correctly handle the case where the main initramfs has the uboot header? (i.e. it's not raw)

Reply:
> Before I go further reviewing this can you please add some documentation inline, so that this is easier to parse later. Especially the alignment concerns.

I'll do

> Also is there areason for the "run loadbootenv" part to be duplicated now?

Uhm, we have now `loadbootenv` exactly as before loading the usual ramdisk with the u-boot header. When `initramfs-append` is detected `loadbootenv_append` is called that keeps using the usual ramdisk (with the u-boot header) and the raw `initramfs-append` file.

> Also, does this correctly handle the case where the main initramfs has the uboot header? (i.e. it's not raw)

Yes, in both case we are going to use the main initramfs with the u-boot header, the difference is that in the append case we call `bootm` from the address `0x13000040`, the offset `0x40` at the LSB is used to skip the header.